### PR TITLE
coqPackages.mathcomp: 1.13.0 -> 1.14.0

### PR DESCRIPTION
**Pull request by proux01**
This is on top of #155635

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by proux01** [comment]
@vbgl successfully tested on coq-nix-toolbox CI: https://github.com/coq-community/coq-nix-toolbox/pull/94

File: pkgs/development/coq-modules/hierarchy-builder/default.nix [hunk 1]
```diff
@@ -5,10 +5,12 @@ with lib; let hb = mkCoqDerivation {
   owner = "math-comp";
   inherit version;
   defaultVersion = with versions; switch coq.coq-version [
+    { case = range "8.15" "8.15"; out = "1.2.1"; }
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/hierarchy-builder/default.nix:8]: I'm wondering if an upper bound is necessary, @gares WDYT?

File: pkgs/development/coq-modules/interval/default.nix [hunk 2]
```diff
@@ -6,11 +6,12 @@ mkCoqDerivation rec {
   domain = "gitlab.inria.fr";
   inherit version;
   defaultVersion = with lib.versions; lib.switch coq.coq-version [
-    { case = range "8.8" "8.14"; out = "4.3.0"; }
+    { case = range "8.8" "8.15"; out = "4.4.0"; }
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/interval/default.nix:9]: Maybe we can leave out the upper bound, like in the opam package (CC @silene)

File: pkgs/development/coq-modules/mathcomp-analysis/default.nix [hunk 3]
```diff
@@ -18,6 +19,7 @@ let mca = mkCoqDerivation {
 
   inherit version;
   defaultVersion = with versions; switch [ coq.version mathcomp.version ]  [
+      { cases = [ (range "8.13" "8.15") (isGe "1.12.0") ];      out = "0.3.13"; }
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/mathcomp-analysis/default.nix:22]: No upper bound actually..

File: pkgs/development/coq-modules/multinomials/default.nix [hunk 4]
```diff
@@ -9,14 +9,15 @@ with lib; mkCoqDerivation {
 
   inherit version;
   defaultVersion =  with versions; switch [ coq.version mathcomp.version ] [
-      { cases = [ (range "8.10" "8.14") (isGe "1.12.0") ];      out = "1.5.4"; }
+      { cases = [ (range "8.10" "8.15") (isGe "1.12.0") ];      out = "1.5.5"; }
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/multinomials/default.nix:12]: same

**Comment by CohenCyril** [comment]
Great!!
Can you bump up extructures to [v0.3.1](https://github.com/arthuraa/extructures/releases/tag/v0.3.1)? Also, I'm going to release odd-order, stay tuned...

**Comment by CohenCyril** [comment]
@proux01 could you copy paste the result of `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` ?

**Comment by proux01** [comment]
41 packages updated:
coq8.13-addition-chains coq8.13-autosubst coq8.13-category-theory coq8.13-compcert coq8.13-compcert coq8.13-coq-bits coq8.13-CoqEAL coq8.13-coquelicot coq8.13-deriving coqPackages.extructures (0.3.0 → 0.3.1) coqPackages.flocq (3.4.2 → 3.4.3) coqPackages.fourcolor (1.2.3 → 1.2.4) coq8.13-gaia coq8.13-gaia-hydras coq8.13-gappalib coq8.13-graph-theory coqPackages.interval (4.3.0 → 4.4.0) coqPackages.mathcomp-algebra (1.12.0 → 1.14.0) coqPackages.mathcomp (1.12.0 → 1.14.0) coqPackages.mathcomp-character (1.12.0 → 1.14.0) coqPackages.mathcomp-field (1.12.0 → 1.14.0) coqPackages.mathcomp-fingroup (1.12.0 → 1.14.0) coqPackages.mathcomp-solvable (1.12.0 → 1.14.0) coqPackages.mathcomp-ssreflect (1.12.0 → 1.14.0) coq8.13-mathcomp1.14-abel coqPackages.mathcomp-analysis (0.3.10 → 0.3.13) coq8.13-mathcomp1.14-bigenough coq8.13-mathcomp1.14-finmap coqPackages.multinomials (1.5.4 → 1.5.5) coq8.13-mathcomp1.14-real-closed coq8.13-mathcomp1.14-tarjan coq8.13-mathcomp1.14-word coq8.13-mathcomp1.14-zify coqPackages.odd-order (1.12.0 → 1.13.0) coq8.13-QuickChick coq8.13-reglang coq8.13-relation-algebra coq8.13-verdi coq8.13-VST frama-c why3

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build coqPackages coqPackages_8_14 coqPackages_8_15

**Review by CohenCyril — APPROVED** [review]
@vbgl ok modulo a build failure on darwin which is defintely not due to us, but which would require fixing:
`builder for '/nix/store/69ls90p7c7yigjvhv3zq9v445q2m7ka4-ocaml4.10.2-cairo2-0.6.2.drv' failed with exit code 1`
cf https://logs.nix.ci/?key=nixos/nixpkgs.157033&attempt_id=fac4be14-8607-45d6-b07d-06ea7c100017)

Please merge

**Comment by proux01** [comment]
Thanks to both of you!

**Pull request merged**